website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,100 @@
+## Usage
+
+### When to use
+
+- When displaying code examples and longer snippets of code that benefit from syntax highlighting.
+
+### When not to use
+
+- As a full-blown code editor
+- When displaying keys, tokens, or other generated values, use a [CopySnippet](/components/copy/snippet) instead.
+
+## Inline
+
+The `isInline` property increases the portability of the CodeBlock to ensure that it can be used in different contexts. For example, a common use case of the CodeBlock is in a "standalone" context, which can be part of a form, multi-step process, and is generally a part of the normal layout flow.
+
+![Code Block with rounded corners in a standalone context](/assets/components/code-block/code-block-rounded-standalone.png)
+
+Sometimes it may be necessary to use the CodeBlock in a more dense layout or nested within another component. In this circumstance, setting `isInline` to true ensures that the CodeBlock fits alongside other elements, in a split view, or as part of a larger layout mechanism.
+
+![Code Block in a block context](/assets/components/code-block/code-block-block-level.png)
+
+## Header
+
+Use a `title` or `description` in the Header to provide additional information, instructions, or to label a CodeBlock. Both of these properties are optional, but including them can help to provide additional context about a specific block of code.
+
+![Example metadata in the Code Block](/assets/components/code-block/code-block-metadata.png)
+
+### When not to use a Header
+
+There can be an overlap between content that you may choose to include in the Header as a `title` or `description`, and content that is part of the normal layout flow in a headline or paragraph. If it is necessary to elevate this content in the hiearchy of the page, we recommend including it in the normal layout flow, rather than as a `title` or `description` within the Code Block.
+
+![An example showcasing the Code Block paired with content in the nautral flow](/assets/components/code-block/code-block-dont-use-metadata.png)
+
+## CopyButton
+
+Use a CopyButton within the CodeBlock to make copying the snippet a single action. More details can be found in the [CopyButton guidelines](/components/copy/button).
+
+![An example of the Copy Button within the Code Block](/assets/components/code-block/code-block-copy-button.png)
```

Comment:
https://github.com/hashicorp/design-system/pull/1691/files#r1367339696

Reply:
Resolved in https://github.com/hashicorp/design-system/pull/1749/commits/19a092530be5f2419739f829a8d000a99a7cb4a7